framework/test/unit/modules/interoperability/mainchain/store.spec.ts

```diff
@@ -381,4 +397,164 @@ describe('Mainchain interoperability store', () => {
+
+			const ccu: CCUpdateParams = {
+				activeValidatorsUpdate: [],
+				certificate: Buffer.alloc(0),
+				inboxUpdate: {
+					crossChainMessages: [],
+					messageWitness: {
+						partnerChainOutboxSize: BigInt(0),
+						siblingHashes: [],
+					},
+					outboxRootWitness: {
+						bitmap: Buffer.alloc(0),
+						siblingHashes: [],
+					},
+				},
+				newCertificateThreshold: BigInt(1),
+				sendingChainID: 2,
+			};
+
+			apiContext = createTransientAPIContext({});
+
+			context = {
+				ccm,
+				ccu,
+				eventQueue: apiContext.eventQueue,
+				feeAddress: Buffer.alloc(0),
+				getAPIContext: jest.fn(() => apiContext),
+				getStore: apiContext.getStore,
+				logger: loggerMock,
+				networkIdentifier: Buffer.alloc(0),
+			};
+
+			beforeCCMSendContext = createCCMsgBeforeSendContext({
+				ccm,
+				eventQueue: context.eventQueue,
+				getAPIContext: context.getAPIContext,
+				logger: context.logger,
+				networkIdentifier: context.networkIdentifier,
+				getStore: context.getStore,
+				feeAddress: EMPTY_FEE_ADDRESS,
+			});
+
+			jest.spyOn(mainchainInteroperabilityStore, 'isLive').mockImplementation();
+			jest.spyOn(mainchainInteroperabilityStore, 'bounce').mockImplementation();
+			jest.spyOn(mainchainInteroperabilityStore, 'sendInternal').mockImplementation();
+			jest
+				.spyOn(mainchainInteroperabilityStore, 'getChainAccount')
+				.mockReturnValue(receivingChainAccount);
+			jest.spyOn(mainchainInteroperabilityStore, 'addToOutbox').mockImplementation();
+			jest.spyOn(mainchainInteroperabilityStore, 'terminateChainInternal').mockImplementation();
+		});
+
+		it('should successfully forward CCM', async () => {
+			receivingChainAccount.status = CHAIN_ACTIVE;
+			jest.spyOn(mainchainInteroperabilityStore, 'isLive').mockResolvedValue(true);
+			jest.spyOn(tokenCCAPI, 'forwardMessageFee').mockResolvedValue(true);
+
+			const result = await mainchainInteroperabilityStore.forward(context);
+			expect(tokenCCAPI.forwardMessageFee).toHaveBeenCalledWith(apiContext, ccm);
+			expect(mainchainInteroperabilityStore.addToOutbox).toHaveBeenCalledWith(
+				receivingChainIDAsStoreKey,
+				ccm,
+			);
+			expect(result).toBe(ForwardResult.SUCCESS);
+		});
+
+		it('should bounce and inform terminated sidechain when sidechain is not active', async () => {
+			const result = await mainchainInteroperabilityStore.forward(context);
+			expect(mainchainInteroperabilityStore.bounce).toHaveBeenCalledWith(ccm);
+			expect(mainchainInteroperabilityStore.sendInternal).toHaveBeenCalled();
+			expect(result).toBe(ForwardResult.INFORM_SIDECHAIN_TERMINATION);
+		});
+
+		it('should throw when tokenCCAPI is not present', async () => {
+			mainchainInteroperabilityStore['interoperableModuleAPIs'].delete(MODULE_ID_TOKEN);
+			await expect(mainchainInteroperabilityStore.forward(context)).rejects.toThrow(
+				'TokenCCAPI does not exist',
+			);
+		});
+
+		it('should throw error when ccm status is not OK', async () => {
```

Comment:
I think the description needs to be changed since it's resolving/returning.

Reply:
Done